File: Components/Strata.Foundation.Core/java/strata/foundation/core/event/IEventSender.java
//////////////////////////////////////////////////////////////////////////////
// IEventSender.java
//////////////////////////////////////////////////////////////////////////////

package strata.foundation.core.event;

import java.util.concurrent.CompletionStage;

public
interface IEventSender<E>
{
    IEventSender<E>
    open() throws Exception;

    IEventSender<E>
    close() throws Exception;

    CompletionStage<SendResult<E>>
    send(E event);

    boolean
    isOpen();

    boolean
    isClosed();
}


//////////////////////////////////////////////////////////////////////////////

File: Components/Strata.Foundation.Core/main/typescript/Event/IEventSender.ts
import {ICompletableSendResult} from "./ICompletableSendResult";

export
interface IEventSender<E>
{
    open(): IEventSender<E>;

    close(): IEventSender<E>;

    send(event: E): ICompletableSendResult<E>;

    isOpen(): boolean;

    isClosed(): boolean;

}